frontend/src/metabase/admin/settings/components/Email/SMTPConnectionForm.tsx

```diff
@@ -53,17 +54,33 @@ type FormValueProps = Pick<
   | "email-smtp-password"
 >;
 
-const FORM_VALUE_SCHEMA = Yup.object({
-  "email-smtp-host": Yup.string().required(Errors.required).default(""),
-  "email-smtp-port": Yup.number()
-    .positive()
-    .nullable()
-    .required(Errors.required)
-    .default(null),
-  "email-smtp-security": Yup.string(),
-  "email-smtp-username": Yup.string().default(""),
-  "email-smtp-password": Yup.string().default(""),
-});
+const anySchema = Yup.mixed().nullable().default(null);
+
+// we need to allow this form to be submitted even when we have removed certain inputs
+// when they are set by env vars
+const getFormValueSchema = (elementMap: _.Dictionary<SettingElement>) => {
+  return Yup.object({
+    "email-smtp-host": elementMap["email-smtp-host"].is_env_setting
+      ? anySchema
+      : Yup.string().required(Errors.required).default(""),
+    "email-smtp-port": elementMap["email-smtp-port"].is_env_setting
+      ? anySchema
+      : Yup.number()
+          .positive()
+          .nullable()
+          .required(Errors.required)
+          .default(null),
+    "email-smtp-security": elementMap["email-smtp-security"].is_env_setting
```

Comment:
This is slightly outside your diff but I believe we need to give Yup localized error messages or it will use its own unlocalized ones